ndk/src/main/jni/utils/stack_unwinder_libunwind.c

```diff
@@ -83,19 +34,69 @@ bsg_libunwind_callback(struct _Unwind_Context *context, void *arg) __asyncsafe {
   return _URC_NO_REASON;
 }
 
+#if defined(__arm__)
+ssize_t
+bsg_unwind_stack_libunwind_arm32(bsg_stackframe stacktrace[BUGSNAG_FRAMES_MAX],
+                                 siginfo_t *info, void *user_context) __asyncsafe {
+  unw_cursor_t cursor;
+  unw_context_t uc;
+  int index = 0;
+
+  unw_getcontext(&uc);
+  unw_init_local(&cursor, &uc);
+  // Initialize cursor state with register data, if any
+  if (user_context != NULL) {
+    const ucontext_t *signal_ucontext = (const ucontext_t *)user_context;
+    const struct sigcontext *signal_mcontext = &(signal_ucontext->uc_mcontext);
+    unw_set_reg(&cursor, UNW_ARM_R0, signal_mcontext->arm_r0);
+    unw_set_reg(&cursor, UNW_ARM_R1, signal_mcontext->arm_r1);
+    unw_set_reg(&cursor, UNW_ARM_R2, signal_mcontext->arm_r2);
+    unw_set_reg(&cursor, UNW_ARM_R3, signal_mcontext->arm_r3);
+    unw_set_reg(&cursor, UNW_ARM_R4, signal_mcontext->arm_r4);
+    unw_set_reg(&cursor, UNW_ARM_R5, signal_mcontext->arm_r5);
+    unw_set_reg(&cursor, UNW_ARM_R6, signal_mcontext->arm_r6);
+    unw_set_reg(&cursor, UNW_ARM_R7, signal_mcontext->arm_r7);
+    unw_set_reg(&cursor, UNW_ARM_R8, signal_mcontext->arm_r8);
+    unw_set_reg(&cursor, UNW_ARM_R9, signal_mcontext->arm_r9);
+    unw_set_reg(&cursor, UNW_ARM_R10, signal_mcontext->arm_r10);
+    unw_set_reg(&cursor, UNW_ARM_R11, signal_mcontext->arm_fp);
+    unw_set_reg(&cursor, UNW_ARM_R12, signal_mcontext->arm_ip);
+    unw_set_reg(&cursor, UNW_ARM_R13, signal_mcontext->arm_sp);
+    unw_set_reg(&cursor, UNW_ARM_R14, signal_mcontext->arm_lr);
+    unw_set_reg(&cursor, UNW_ARM_R15, signal_mcontext->arm_pc);
+    unw_set_reg(&cursor, UNW_REG_IP, signal_mcontext->arm_pc);
+    unw_set_reg(&cursor, UNW_REG_SP, signal_mcontext->arm_sp);
```

Comment:
The previous implementation had a block comment that was fairly helpful at understanding what this code was doing - I'm having a harder time following this method without that sort of documentation. 

It's not clear to me why we now need to set the register values for `UNW_REG_IP` and `UNW_REG_SP`, and why the single-frame mode has been removed if there's a risk of unwinding causing a segfault.

Reply:
> It's not clear to me why we now need to set the register values for UNW_REG_IP and UNW_REG_SP,

This is now using a different library which provides the option of setting additional registers for IP and SP, so I set those as well. 

> if there’s a risk of unwinding causing a segfault

There’s no longer a risk so it’s not needed. The underlying implementation is more stable.